Plugins/Public/contraband/Main.cpp

```diff
@@ -0,0 +1,371 @@
+							lawsLoaded++;
+						}
+					}
+				}
+			}
+		}
+		ini.close();
+
+		ConPrint(L"LAWS: Houses Loaded: %u \n", housesLoaded);
+		ConPrint(L"LAWS: House Systems Loaded: %u \n", houseSystemsLoaded);
+		ConPrint(L"LAWS: Laws Loaded: %u \n", lawsLoaded);
+		ConPrint(L"LAWS: Contraband Items Loaded: %u \n", contrabandLoaded);
+	}
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Functions
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void ContrabandWarning(uint iClientID, wstring wscText, ...) // WE WANT BOLD RED
+{
+	wchar_t wszBuf[1024 * 8] = L"";
+	va_list marker;
+	va_start(marker, wscText);
+	_vsnwprintf(wszBuf, sizeof(wszBuf) - 1, wscText.c_str(), marker);
+
+	wstring wscXML = L"<TRA data=\"65281\" mask=\"-1\"/><TEXT>" + XMLText(wszBuf) + L"</TEXT>";
+	HkFMsg(iClientID, wscXML);
+}
+
+static void CheckCargo(int iClientID)
+{
+	if (set_lawsContrabandEnabled)
+	{
+		uint iSystemID;
+		pub::Player::GetSystem(iClientID, iSystemID);
+
+		int iHoldSize;
+		list<CARGO_INFO> lstCargo;
+		HkEnumCargo((const wchar_t*)Players.GetActiveCharacterName(iClientID), lstCargo, iHoldSize);
+
+		for (map<uint, string>::iterator iter = mapHouseAndSystems.begin(); iter != mapHouseAndSystems.end(); iter++)
+		{
+
+			uint currentSector = iter->first;
+			string currentHouse = iter->second;
+
+			if (currentSector == iSystemID)
+			{
+				for (map<string, list<uint>>::iterator iter = mapHouseCargoList.begin(); iter != mapHouseCargoList.end(); iter++)
+				{
+					string currentHouseCompare = iter->first;
+					list<uint> cfgCargoLst = iter->second;
+
+					if (currentHouse == currentHouseCompare)
+					{
+						for (list<CARGO_INFO>::iterator i = lstCargo.begin(); i != lstCargo.end(); ++i)
+						{
+							uint correctCargo = i->iArchID;
+							foreach(cfgCargoLst, uint, cfgCargoIter)
+							{
+								if (*cfgCargoIter == correctCargo) // If their cargo is on the list
+								{
+									ContrabandWarning(iClientID, L"WARNING: You are carrying items which are illegal in this sector.");
+									return;
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+//////////////////////////////////////////////
+//                 Commands                 //
+//////////////////////////////////////////////
+bool UserCmd_ManualCargoCheck(uint iClientID, const wstring & wscCmd, const wstring & wscParam, const wchar_t * usage)
+{
+	CheckCargo(iClientID);
+	return true;
+}
+
+bool UserCmd_Laws(uint iClientID, const wstring & wscCmd, const wstring & wscParam, const wchar_t * usage)
+{
+	if (set_lawsEnabled)
+	{
+		for (map<uint, string>::iterator iter = mapHouseAndSystems.begin(); iter != mapHouseAndSystems.end(); iter++)
+		{
+			uint currentSystem = iter->first;
+			string currentHouse = iter->second;
+
+			uint getSystem;
+			pub::Player::GetSystem(iClientID, getSystem);
+
+			if (currentSystem == getSystem)
```

Comment:
this code looks like it could be replaced by a simple lookup of whether getSystem is a key in mapHouseAndSystems, if so then currentHouse = mapHouseAndSystems[getSystem]